bagua/torch_api/distributed.py

```diff
@@ -199,7 +201,9 @@ def with_bagua(  # pytype: disable=module-attr
             self, "_ddp_params_and_buffers_to_ignore"
         ):  # for compatibility with PyTorch DDP
             self.parameters_to_ignore.extend(self._ddp_params_and_buffers_to_ignore)
-        self.bagua_train_step_counter = 0
+
+        self.communication_interval = communication_interval
+        self.bagua_train_step_counter = -1
```

Comment:
why start with -1

Reply:
> why start with -1

this `bagua_train_step_counter` is increased in pre-forward hook, initialize it to be -1, to make sure the first step will not be skipped.

Reply:
that sound hacky.

Let's just remove changes to this file and move all communication interval logic to the algorithm itself